tests/integration/test_session.py

```diff
@@ -236,3 +236,93 @@ def test_session_aggregates_invalid_release(
     )
 
     sessions_consumer.assert_empty()
+
+
+def test_session_filtering(mini_sentry, relay_with_processing, sessions_consumer):
+    relay = relay_with_processing()
+    sessions_consumer = sessions_consumer()
+
+    project_id = 42
+    project_config = mini_sentry.add_full_project_config(project_id)
+    filter_settings = project_config["config"]["filterSettings"]
+    filter_settings["webCrawlers"] = {"isEnabled": True}
+    filter_settings["clientIps"] = {"blacklistedIps": ["1.2.3.0/24"]}
+    filter_settings["releases"] = {"releases": ["sentry-bad*"]}
+
+    timestamp = datetime.now(tz=timezone.utc)
+    relay.send_session(
+        project_id,
+        {
+            "sid": "8333339f-5675-4f89-a9a0-1c935255ab58",
+            "timestamp": timestamp.isoformat(),
+            "started": timestamp.isoformat(),
+            "attrs": {"release": "sentry-test@1.0.0", "user_agent": "BingBot"},
+        },
+    )
+
+    relay.send_session(
+        project_id,
+        {
+            "sid": "8333339f-5675-4f89-a9a0-1c935255ab59",
+            "timestamp": timestamp.isoformat(),
+            "started": timestamp.isoformat(),
+            "attrs": {"release": "sentry-test@1.0.0", "ip_address": "1.2.3.0/24"},
```

Comment:
Does this IP filtering match the one we have for errors? I think there was something like, that we do not filter on user ips with that denylist, but actual ingestion IPs. We should follow the same here, if it filters on the user ip, this would be a different behaviour.